### PR TITLE
Rust dockerfile update

### PR DESCRIPTION
**Pull request by gregwebs**
These changes except for the formatter were necessary for me to build the dockerfile.
Successfully building `rust_dockerfile` was my test. I also ran `make format` and `make bench` in the container without error.

### CODE REVIEW

File: docker/rust_dockerfile [hunk 1]
```diff
@@ -39,14 +41,14 @@ RUN go get golang.org/x/tools/go/gcimporter15 \
 
 RUN curl https://sh.rustup.rs -sSf | sh -s -- --no-modify-path --default-toolchain nightly-2018-01-12 -y \
 	&& /root/.cargo/bin/rustup default nightly-2018-01-12 \
-	&& /root/.cargo/bin/cargo install --vers 0.3.4 --force rustfmt-nightly
+	&& /root/.cargo/bin/rustup component add rustfmt-preview
```
Comment (siddontang) [inline docker/rust_dockerfile]: we forcibly use nightly 0.3.4 version for rustfmt here

**Comment by siddontang** [comment]
Thank you @gregwebs

File: docker/rust_dockerfile [hunk 2]
```diff
@@ -8,11 +8,13 @@ RUN yum install -y epel-release
 RUN yum makecache \
 	&& yum update -y \
 	&& yum install -y tar wget git which file unzip python-pip \
-	&& yum install -y make cmake gcc gcc-c++ libstdc++-static pkg-config psmisc \
+	&& yum install -y make cmake3 gcc gcc-c++ libstdc++-static pkg-config psmisc \
 	&& yum install -y libzstd-devel lz4-devel zlib-devel bzip2-devel snappy-devel libdwarf-devel elfutils-libelf-devel elfutils-devel binutils-devel \
```
Comment (siddontang) [inline docker/rust_dockerfile]: seem we can remove libzstd-devel lz4-devel zlib-devel bzip2-devel snappy-devel now

File: docker/rust_dockerfile (hunk 1 above)
Comment (gregwebs) [reply docker/rust_dockerfile]: Do you want me to remove this commit from my PR or do you want to make a new PR with cherry-picks from this branch?

Comment (siddontang) [reply docker/rust_dockerfile]: I think we should keep the origin, still use the 0.3.4.

**Comment by gregwebs** [comment]
I removed a couple of my commits, and added two more (one is the suggested library removal).

**Comment by siddontang** [comment]
Thanks @gregwebs 

But I can't use your dockerfile to build TiKV, meet following error:

```
   Compiling rocksdb v0.3.0 (https://github.com/pingcap/rust-rocksdb.git#cb63dcd1)
   Compiling kvproto v0.0.1 (https://github.com/pingcap/kvproto.git#6bcc71d6)
error: Could not compile `tikv`.
warning: build failed, waiting for other jobs to finish...
```

Do you meet this problem?

**Comment by gregwebs** [comment]
I did not encounter that problem, but let me try destroying my cached docker images and then I can share how I am running this.

**Comment by siddontang** [comment]
@gregwebs 
Seem this can only occur on Mac OSX, do you use Mac OSX?

**Comment by gregwebs** [comment]
I am on Linux, still working fine after re-building the docker image from scratch.

I run `just setup-docker && just docker make all` from here:
https://github.com/gregwebs/tikv/compare/rust_dockerfile-update...docker-runner

I can also PR this running code according to how you want to organize your project.

**Comment by siddontang** [comment]
Well, I think we will dive to see why it fails on Mac OSX later. 

PTAL @iamxy

**Comment by siddontang** [comment]
Hi @gregwebs

I increase my Mac OSX docker memory limitation from 2 GB to 8 GB and now it can compile successfully, thank you.

LGTM, PTLA @iamxy

**Comment by iamxy** [comment]
LGTM
Thanks @gregwebs

**Pull request merged**